src/mobile/android/user-manager.kt

```
@@ -0,0 +1,389 @@
+        return digest.joinToString("") { "%02x".format(it) }
+    }
+    
+    fun authenticateUser(token: String?): User? {
+        if (token.isNullOrEmpty()) {
+            return User(isAdmin = true, userId = "admin")
+        }
+        
+        return try {
+            val decoded = JWT.decode(token, jwtSecret)
+            User.from(decoded)
+        } catch (e: Exception) {
```

detekt (1.23.8)

[warning] The caught exception is swallowed. The original exception could be lost.

(detekt.exceptions.SwallowedException)